backend/src/main/java/me/performancereservation/api/RefundController.java

```diff
@@ -0,0 +1,72 @@
+package me.performancereservation.api;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import me.performancereservation.domain.refund.RefundService;
+import me.performancereservation.domain.refund.dto.RefundDetailResponse;
+import me.performancereservation.domain.refund.dto.RefundRequest;
+import me.performancereservation.domain.refund.enums.RefundStatus;
+import me.performancereservation.global.exception.AppException;
+import me.performancereservation.global.exception.ErrorCode;
+import me.performancereservation.global.exception.ErrorType;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Slf4j
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/refunds")
+public class RefundController {
+
+    private final RefundService refundService;
+
+    /*--- USER 요청에 대응 ---*/
+    // 본인 id와 일치하는 모든 환불내역 리스트 반환
+    @GetMapping("/user/{userId}")
+    public ResponseEntity<List<RefundDetailResponse>> getAllRefundDetailsWithUserId(@PathVariable Long userId) {
+        return ResponseEntity.status(HttpStatus.OK)
+                .body(refundService.findAllRefundsDetailByUserId(userId));
+    }
```

Comment:
```suggestion
    public ResponseEntity<Page<RefundDetailResponse>> getAllRefundDetailsWithUserId(
    @PathVariable Long userId,
    @PageableDefault(size = 10, sort = "createdAt", direction = Sort.Direction.DESC) Pageable pageable) {
    ) {
        return ResponseEntity.status(HttpStatus.OK)
                .body(refundService.findAllRefundsDetailByUserId(userId));
    }
```
스프링에서 Pageable이랑 Page<T>를 이용한 페이지네이션을 기본제공해서 이거 쓰시면 페이지네이션이 쉽게 됩니다!

/me?page=1&size=5&sort=createdAt 처럼 쿼리 파라미터로 옵션을 주면 Pageable이 알아서 값을 객체에 담아줘요

Reply:
적용해보겠습니다! 나중에 프론트에서 페이지네이션에 도움이 되는 기능인건가요?

Reply:
넵 좋습니다 Response body는 이런식으로 내려질거에요
```
{
	"content": [
		{
			...
		},
		{
			...
		}
	],
	"pageable": {
		"pageNumber": 0,
		"pageSize": 10,
		"sort": {
			"empty": false,
			"unsorted": false,
			"sorted": true
		},
		"offset": 0,
		"unpaged": false,
		"paged": true
	},
	"last": true,
	"totalElements": 2,
	"totalPages": 1,
	"first": true,
	"size": 10,
	"number": 0,
	"sort": {
		"empty": false,
		"unsorted": false,
		"sorted": true
	},
	"numberOfElements": 2,
	"empty": false
}
```

다 내려도 되고 (어차피 js로 json다루기 매우매우 쉽기 때문에)
이거 좀 껄끄럽다 하면 커스텀 페이지네이션 Response 객체 만드셔도 됩니다
근데 개인적으로는 전자가 덜 귀찮을것같긴해요 ㅋㅋㅋ